### DISCUSSION TITLE
An awaitable Blazor Modal Dialog

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#122] [General]
# An awaitable Blazor Modal Dialog

I already showcased a simple modal dialog in a previous post. However, the dialog was not awaitable. In this post, I'll show you how to create an awaitable modal dialog.

https://steven-giesel.com/blogPost/79ab37d4-5fdf-4c36-8b99-ac17bbe1fc01

**Comment by paulvanbladel** [comment]
Nice !
Isn't UploadFileModalDialogObject missing in your sample?

**Reply by linkdotnet** [reply]
Indeed it is - but imagine it as a placeholder for whatever you wanna return.
Basically it is literally from this blog, and the source to that file is: https://github.com/linkdotnet/Blog/blob/master/src/LinkDotNet.Blog.Web/Features/Components/UploadFileModalDialogObject.cs
